index.ts

```diff
@@ -18,6 +18,13 @@ declare namespace ethereum {
 /** Host IPFS interface */
 export declare namespace ipfs {
   function cat(hash: string): Bytes | null
+  function map(hash: String, callback: String, flags: String[]): void
+}
+
+export namespace ipfs {
+    export function mapJSON(hash: String, callback: String): void {
+        map(hash, callback, ["json"])
```

Comment:
Did you test this in a subgraph to see if this compiles and runs?

Reply:
Yes, I tested it with the [ipfs-map](https://github.com/lutter/dice2win-subgraph/tree/ipfs-map) branch of dice2win